polkadot/xcm/pallet-xcm/src/precompiles/IXcm.sol

```diff
@@ -1,26 +1,28 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.20;
 
-/// @dev The XCM Precompile contract address.
+/// @dev The on-chain address of the XCM (Cross-Consensus Messaging) precompile.
 address constant XCM_PRECOMPILE_ADDRESS = address(0xA0000);
 
-/// @title An interface for interacting with `pallet_xcm`
-/// @notice Provides functions for executing and sending XCM messages.
-/// Learn more about XCM: https://docs.polkadot.com/develop/interoperability/
-/// @dev All parameters must be encoded using the SCALE codec.
+/// @title XCM Precompile Interface
+/// @notice A low-level interface for interacting with `pallet_xcm`.
+/// It forwards calls directly to the corresponding dispatchable functions, providing access to XCM execution and message passing.
+/// Learn more about XCM: https://docs.polkadot.com/develop/interoperability
+/// @dev All parameters must be encoded using the SCALE codec. See https://docs.polkadot.com/polkadot-protocol/parachain-basics/data-encoding
 interface IXcm {
-    /// @notice Weight v2
+    /// @notice Weight v2. See https://docs.polkadot.com/polkadot-protocol/parachain-basics/blocks-transactions-fees/fees/#transactions-weights-and-fees
     struct Weight {
-        /// The computational time used to execute some logic based on reference hardware.
+        /// @custom:property The computational time used to execute some logic based on reference hardware.
         uint64 refTime;
-        /// The size of the proof needed to execute some logic.
+        /// @custom:property The size of the proof needed to execute some logic.
```

Comment:
Nit: The same for proofSize we could add context to what this proof is, not in detail but just a simple explanation the docs [here](https://contracts.polkadot.io/differences_to_eth) have a good explanation we could take inspiration from